File: python/3228.maximum-number-of-operations-to-move-ones-to-the-end.py
class Solution:
    def maxOperations(self, s: str) -> int:
        inc = 0
        cnt = 0
        prev = ""

        for i in range(len(s) - 1, -1, -1):
            if prev != "0" and s[i] == "0":
                inc += 1

            elif s[i] == "1":
                cnt += inc

            prev = s[i]

        return cnt

File: cpp/3228.maximum-number-of-operations-to-move-ones-to-the-end.cpp
#include <string>
using std::string;

class Solution {
public:
  int maxOperations(string s) {
    int inc = 0, cnt = 0;
    char p = '1';

    for (int i = s.size(); i >= 0; i--) {
      if (p != '0' && s[i] == '0') {
        inc++;
      } else if (s[i] == '1') {
        cnt += inc;
      }

      p = s[i];
    }

    return cnt;
  }
};